diff --git a/mujoco_py/builder.py b/mujoco_py/builder.py
index 5280b2c..3692908 100644
--- a/mujoco_py/builder.py
+++ b/mujoco_py/builder.py
@@ -89,6 +89,15 @@ The easy solution is to `import mujoco_py` _before_ `import glfw`.
     builder = Builder(mujoco_path)
     cext_so_path = builder.get_so_file_path()
 
+    # https://github.com/openai/mujoco-py/issues/523#issuecomment-651445813
+    # Handle issues with building in read-only containers
+    # Check if we have write access to the cext_so_path.
+    # If not, it's probably because mujoco-py has been installed and everything is
+    # read-only. Returning here is necessary because the lock creation will fail.
+    # It might be better to try-catch the lock but this minimizes the diff complexity.
+    if not os.access(os.path.dirname(cext_so_path), os.W_OK):
+        return load_dynamic_ext('cymj', cext_so_path)
+
     lockpath = os.path.join(os.path.dirname(cext_so_path), 'mujocopy-buildlock')
 
     with fasteners.InterProcessLock(lockpath):
